DllImportGenerator/DllImportGenerator/Marshalling/BlittableArrayMarshaller.cs

```diff
@@ -56,6 +56,44 @@ public override IEnumerable<StatementSyntax> Generate(TypePositionInfo info, Stu
             var (managedIdentifer, nativeIdentifier) = context.GetIdentifiers(info);
             if (!info.IsByRef && !info.IsManagedReturnPosition && context.PinningSupported)
             {
+                string byRefIdentifier = $"__byref_{managedIdentifer}";
+                if (context.CurrentStage == StubCodeContext.Stage.Marshal)
+                {
+                    // ref <elementType> <byRefIdentifier> = <managedIdentifer> == null ? ref Unsafe.NullRef<<elementType>>() : ref MemoryMarshal.GetArrayDataReference(<managedIdentifer>);
```

Comment:
We should add an explicit comment about the non-null logic here and that it is an explicit decision for backwards compat and supporting API behavior. I think we typically document changes but this behavior is unusual and non-obvious enough that we should document this as an explicit decision.